Automated Smoketest 
Sha: a9e2e0f6f6d3b803acf006f2030ccd528a7a470e
Build: failed
Number of Error(s): 3
/mnt/disk1/home/vamosax/smoketest/PR-9089/HPCC-Platform/thorlcr/thorutil/thbuf.cpp:753:14: error: \xe2\x80\x98reuseRowSets\xe2\x80\x99 was not declared in this scope
/mnt/disk1/home/vamosax/smoketest/PR-9089/HPCC-Platform/thorlcr/thorutil/thbuf.cpp:1067:9: error: \xe2\x80\x98reuseRowSets\xe2\x80\x99 was not declared in this scope
/mnt/disk1/home/vamosax/smoketest/PR-9089/HPCC-Platform/thorlcr/thorutil/thbuf.cpp:1071:9: error: \xe2\x80\x98reuseRowSets\xe2\x80\x99 was not declared in this scope

--- a/thorlcr/thorutil/thbuf.cpp
+++ b/thorlcr/thorutil/thbuf.cpp
@@ -750,6 +750,8 @@ class CSharedWriteAheadBase : public CSimpleInterface, implements ISharedSmartBu
 
     void reuse(CRowSet *rowset)
     {
+        if (!reuseRowSets)
+            return;
         rowset->clear();
         CriticalBlock b(rowSetCacheCrit);
         if (cachedRowSets.ordinality() < (outputs.ordinality()*2))
@@ -910,6 +912,7 @@ protected:
     Linked<IOutputMetaData> meta;
     QueueOf<CRowSet, false> chunkPool;
     unsigned maxPoolChunks;
+    bool reuseRowSets;
 
     inline const rowcount_t readerWait(COutput &output, const rowcount_t rowsRead)
     {
@@ -1062,9 +1065,11 @@ public:
         }
         inMemRows.setown(newRowSet(0));
         maxPoolChunks = MIN_POOL_CHUNKS;
+        reuseRowSets = true;
     }
     ~CSharedWriteAheadBase()
     {
+        reuseRowSets = false;
         ForEachItemIn(o, outputs)
         {
             COutput &output = queryCOutput(o);
@@ -1072,7 +1077,6 @@ public:
         }
         inMemRows.clear();
     }
-
 
     unsigned anyReaderBehind()
     {

